packages/admin/src/targetGroups/TargetGroupDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import { useApolloClient } from "@apollo/client";
+import { Field, FinalForm, FinalFormInput, FinalFormSubmitEvent, MainContent, useFormApiRef, useStackSwitchApi } from "@comet/admin";
+import { AddFilled, Close } from "@comet/admin-icons";
+import { ContentScopeInterface } from "@comet/cms-admin";
+import { Button, Card, Dialog, DialogActions, DialogContent, DialogTitle, IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createTargetGroupMutation } from "./TargetGroupDialog.gql";
+import { GQLCreateTargetGroupMutation, GQLCreateTargetGroupMutationVariables } from "./TargetGroupDialog.gql.generated";
+import { EditTargetGroupFinalFormValues } from "./TargetGroupForm";
+
+interface FormProps {
+    open: boolean;
+    handleClose: () => void;
+    scope: ContentScopeInterface;
+}
+
+export function TargetGroupDialog({ scope, open, handleClose }: FormProps): React.ReactElement {
+    const client = useApolloClient();
+    const mode = "add";
+    const formApiRef = useFormApiRef<EditTargetGroupFinalFormValues>();
+    const stackSwitchApi = useStackSwitchApi();
+
+    const handleSubmit = async (
+        state: EditTargetGroupFinalFormValues,
+        form: FormApi<EditTargetGroupFinalFormValues>,
+        event: FinalFormSubmitEvent,
+    ) => {
+        const output = {
+            ...state,
+        };
+
+        const { data: mutationResponse } = await client.mutate<GQLCreateTargetGroupMutation, GQLCreateTargetGroupMutationVariables>({
+            mutation: createTargetGroupMutation,
+            variables: { scope, input: output },
+        });
+        if (!event.navigatingBack) {
+            const id = mutationResponse?.createTargetGroup.id;
+            if (id) {
+                setTimeout(() => {
+                    stackSwitchApi.activatePage("edit", id);
+                });
+            }
+        }
+    };
+    return (
+        <Dialog open={open} onClose={handleClose} maxWidth="sm">
```

Comment:
Comet has an EditDialog component which could be used here 
https://comet-admin.netlify.app/?path=/story/stories-components-editdialog-edit-dialog--edit-dialog